File: src/MyServiceBus/Topology/SagaStateMachineTopology.cs
using System.Text.Json.Serialization;
using MyServiceBus.Orchestration;

namespace MyServiceBus.Topology;

/// <summary>
/// Describes a registered saga state machine and its receive endpoint.
/// </summary>
public sealed record SagaStateMachineTopology(
    [property: JsonPropertyName("definition")] SagaStateMachineDefinition Definition,
    [property: JsonPropertyName("endpointName")] string EndpointName);

File: src/Java/myservicebus/src/main/java/com/myservicebus/topology/SagaStateMachineTopology.java
package com.myservicebus.topology;

import com.myservicebus.orchestration.SagaStateMachineDefinition;

/** Describes a registered saga state machine and its receive endpoint. */
public record SagaStateMachineTopology(
        SagaStateMachineDefinition definition,
        String endpointName) {
}
